src/vscode-bicep/src/commands/deploy.ts

```diff
@@ -124,6 +124,12 @@ export class DeployCommand implements Command {
         context
       );
 
+      const options = {
+        title: `Please enter name for deployment`,
+        value: "bicep_deployment_".concat(Date.now().toString()),
```

Comment:
why not using "bicep_deployment_" + DateTime.UtcNow.ToString("yyyyMMddHHmmss") ?

Reply:
Also, can we add the bicep filename to start of this (trimmed and escaped if needed)?

Reply:
I updated to specify 'YYYYMMDDHHmmss' format. I checked az cli experience and looks like it just uses the file name for deployment. The changes in this PR match the existing behavior, except an option to accept name from the user. 
I would like to keep this as is and wait for customer feedback. @StephenWeatherford, do you have any concerns?

Reply:
PC, Stephen and I discussed this offline and came up with <bicep_filename>_timestamp.

@bmoore-msft, could you please provide some input? I am trying to add support to accept deployment name from the user. By default, we want to prefil the input box with <bicep_filename>_timestamp. User can either use it or update the name. Does <bicep_filename>_timestamp sound like a good idea? Should we append timestamp to user provided input for uniqueness?

cc: @StephenWeatherford, @puicchan

Reply:
This is for deploying directly from VSCode?

I think the main thing when deploying from the dev env is "debugging" so I would worry less about uninqueness (e.g. a really long timestamp) and more about ease of getting the results if needed.  So might go with something like `<filename>-MMDD-HHMM`

but in general I think looks good.

Reply:
Thanks for the input @bmoore-msft. Yes, this is for deploying directly from VSCode.

I'll go ahead with <filename>-MMDD-HHMM then. @StephenWeatherford, @puicchan, could you please +1?

Reply:
Looks good to me, or else <filename>-YYMMDD-HHMM         (added year)

You'll need to remove or substitute invalid characters from filename and trim if too long.  Don't know max deployment name length.

Reply:
64 chars is the limit